cmd/end2endtest/dynamicensus.go

```diff
@@ -0,0 +1,320 @@
+			c.nvotes, time.Since(startTime), int(float64(c.nvotes)/time.Since(startTime).Seconds()))
+
+		// Set the account back to the organization account
+		if err := api.SetAccount(c.accountPrivKeys[0]); err != nil {
+			errCh <- fmt.Errorf("unexpected error when set the account, %s", err)
+			return
+		}
+
+		// End the election by setting the status to ENDED
+		log.Infof("ending election...")
+		if _, err := api.SetElectionStatus(electionID, "ENDED"); err != nil {
+			errCh <- fmt.Errorf("unexpected error when ended the elecion, %s", err)
+			return
+		}
+
+		// Wait for the election to be in RESULTS state
+		ctx, cancel = context.WithTimeout(context.Background(), time.Second*100)
+		defer cancel()
+		election, err := api.WaitUntilElectionResults(ctx, electionID)
+		if err != nil {
+			errCh <- fmt.Errorf("unexpected error when wait until results, %s", err)
+			return
+		}
+
+		expectedResults := [][]*types.BigInt{votesToBigInt(uint64(c.nvotes-3)*10, 10, 0)}
+
+		// only the first overwrite should be valid in the results and must math with the expected results
+		if !matchResults(election.Results, expectedResults) {
+			errCh <- fmt.Errorf("election result must match, expected Results: %s but got Results: %v", expectedResults, election.Results)
+			return
+		}
+
+		t.elections[0].election, err = api.Election(electionID)
+		if err != nil {
+			errCh <- fmt.Errorf("unexpected err when retrieve election details, %s", err)
+			return
+		}
+
+		log.Infof("election %s status is RESULTS", electionID.String())
+		log.Infof("election results: %v %x %s", election.Results, election.CensusRoot, t.elections[0].election.Census.CensusURL)
+
+	}()
+
+	// election with dynamic census disabled
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		api := t.elections[1].api
+		electionID := t.elections[1].election.ElectionID
+
+		log.Infof("enqueuing %d votes", len(t.elections[1].voterAccounts[1:]))
+		votes := []*apiclient.VoteData{}
+		for _, acct := range t.elections[1].voterAccounts[1:] {
+			votes = append(votes, &apiclient.VoteData{
+				ElectionID:   electionID,
+				ProofMkTree:  t.elections[1].proofs[acct.Address().Hex()],
+				Choices:      []int{0},
+				VoterAccount: acct,
+			})
+		}
+		t.elections[0].sendVotes(votes)
+
+		var censusRoot2 types.HexBytes
+		var err error
+
+		if censusRoot2, _, err = setupNewCensusAndVote(t.elections[1]); err != nil {
+			errCh <- fmt.Errorf("unexpected error from electionID: %s %s", electionID, err)
+			return
+		}
+
+		log.Debugf("election details before: %s %s %x", t.elections[1].election.Census.CensusOrigin, t.elections[1].election.Census.CensusURL, t.elections[1].election.Census.CensusRoot)
+
+		if _, err := api.TransactionSetCensus(electionID, vapi.ElectionCensus{
+			CensusOrigin: "OFF_CHAIN_TREE_WEIGHTED",
+			CensusRoot:   censusRoot2,
+			CensusURL:    "http://test/census",
+		}); err != nil {
+			log.Debug("expected error from election without dynamic census enabled", err)
```

Comment:
please check for the specific error you're expecting. if the error is not what you expect, fail the test